openedx/core/lib/django_courseware_routers.py

```diff
@@ -45,13 +45,15 @@ def allow_relation(self, obj1, obj2, **hints):  # pylint: disable=unused-argumen
             return False
         return None
 
-    def allow_migrate(self, db, model):  # pylint: disable=unused-argument
+    def allow_migrate(self, db, app_label, model_name=None, **hints):  # pylint: disable=unused-argument
         """
         Only sync StudentModuleHistoryExtended to StudentModuleHistoryExtendedRouter.DATABASE_Name
         """
-        if self._is_csmh(model):
-            return db == self.DATABASE_NAME
-        elif db == self.DATABASE_NAME:
+        if model_name is not None:
+            model = hints.get('model')
```

Comment:
@bdero I see what you mean.  I'm not sure why I didn't see this during installation, but I can make this more robust.